ofono2mm/mm_modem.py

```diff
@@ -120,13 +128,122 @@ async def init_mm_3gpp_interface(self):
+                mm_bearer_interface = MMBearerInterface(self.index, self.bus, self.ofono_proxy, self.modem_name, self.ofono_modem, self.ofono_props, self.ofono_interfaces, self.ofono_interface_props, self)
+                ip_method = 0
+                if 'Method' in ctx[1]['Settings'].value:
+                    if ctx[1]['Settings'].value['Method'].value == "static":
+                        ip_method = 2
+                    if ctx[1]['Settings'].value['Method'].value == "dhcp":
+                        ip_method = 3
+                ip_address = ''
+                if 'Address' in ctx[1]['Settings'].value:
+                    ip_address = ctx[1]['Settings'].value['Address'].value
+                ip_dns = []
+                if 'DomainNameServers' in ctx[1]['Settings'].value:
+                    ip_dns = ctx[1]['Settings'].value['DomainNameServers'].value
+                ip_gateway = ''
+                if 'Gateway' in ctx[1]['Settings'].value:
+                    ip_gateway = ctx[1]['Settings'].value['Gateway'].value
+                mm_bearer_interface.props.update({
+                    "Interface": ctx[1]['Settings'].value['Interface'] if 'Interface' in ctx[1]['Settings'].value else Variant('s', ''),
+                    "Connected": ctx[1]['Active'],
+                    "Ip4Config": Variant('a{sv}', {
+                        "method": Variant('u', ip_method),
+                        "dns1": Variant('s', ip_dns[0] if len(ip_dns) > 0 else ''),
+                        "dns2": Variant('s', ip_dns[1] if len(ip_dns) > 1 else ''),
+                        "dns3": Variant('s', ip_dns[2] if len(ip_dns) > 2 else ''),
+                        "gateway": Variant('s', ip_gateway)
+                    }),
+                    "Properties": Variant('a{sv}', {
+                        "apn": ctx[1]['AccessPointName']
+                    })
+                })
+                if 'Interface' in ctx[1]['Settings'].value:
+                    self.props['Ports'].value.append([ctx[1]['Settings'].value['Interface'].value, 2])
+                    self.emit_properties_changed({'Ports': self.props['Ports'].value})
+                with open('/usr/lib/ofono2mm/ofono_context.xml', 'r') as f:
+                    ctx_introspection = f.read()
+                ofono_ctx_object = self.bus.get_proxy_object('org.ofono', ctx[0], ctx_introspection)
+                ofono_ctx_interface = ofono_ctx_object.get_interface('org.ofono.ConnectionContext')
+                ofono_ctx_interface.on_property_changed(mm_bearer_interface.ofono_context_changed)
+                mm_bearer_interface.ofono_ctx = ctx[0]
+                self.bus.export('/org/freedesktop/ModemManager/Bearer/' + str(bearer_i), mm_bearer_interface)
+                self.props['Bearers'].value.append('/org/freedesktop/ModemManager/Bearer/' + str(bearer_i))
+                self.bearers['/org/freedesktop/ModemManager/Bearer/' + str(bearer_i)] = mm_bearer_interface
```

Comment:
As above, you can use built-in python features to avoid explicitly casting the integer to a string